From 438eaa1c73d861bc95690cbd71ad0b53b6962172 Mon Sep 17 00:00:00 2001
From: b02550 <b02550@freescale.com>
Date: Fri, 12 Jun 2009 14:14:18 +0800
Subject: [PATCH] ENGR00113242 IPUv3: Add new ioctl to get ipu channel type of fb.

Add new ioctl to get ipu channel type to avoid the conflict between
3Stack and BBG board.

Signed-off-by: DingQiang <b02550@freescale.com>
---
 drivers/video/mxc/mxc_ipuv3_fb.c |    9 +++++++++
 include/linux/mxcfb.h            |    1 +
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index e1a1707..be744ed 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -544,6 +544,15 @@ static int mxcfb_ioctl(struct fb_info *fbi, unsigned int cmd, unsigned long arg)
 							 pos.x, pos.y);
 			break;
 		}
+	case MXCFB_GET_FB_IPU_CHAN:
+		{
+			struct mxcfb_info *mxc_fbi =
+				(struct mxcfb_info *)fbi->par;
+
+			if (put_user(mxc_fbi->ipu_ch, argp))
+				return -EFAULT;
+			break;
+		}
 	default:
 		retval = -EINVAL;
 	}
diff --git a/include/linux/mxcfb.h b/include/linux/mxcfb.h
index c7bc6bc..f54a2c0 100644
--- a/include/linux/mxcfb.h
+++ b/include/linux/mxcfb.h
@@ -49,6 +49,7 @@ struct mxcfb_pos {
 #define MXCFB_SET_GBL_ALPHA     _IOW('F', 0x21, struct mxcfb_gbl_alpha)
 #define MXCFB_SET_CLR_KEY       _IOW('F', 0x22, struct mxcfb_color_key)
 #define MXCFB_SET_OVERLAY_POS   _IOW('F', 0x24, struct mxcfb_pos)
+#define MXCFB_GET_FB_IPU_CHAN   _IOR('F', 0x25, u_int32_t)
 
 #ifdef __KERNEL__
 
-- 
1.5.4.4

